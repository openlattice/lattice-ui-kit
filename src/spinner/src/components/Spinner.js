import styled, { keyframes } from 'styled-components';

const spin = keyframes`
  0% {
    transform: rotate(0);
  }

  100% {
    transform: rotate(360deg);
  }
`;

const Spinner = styled.div`
  &::before {
    animation: ${spin} 0.75s linear infinite;
    border: 1px solid #c5d5e5;
    border-radius: 50%;
    border-top-color: #135;
    box-sizing: border-box;
    content: '';
    top: 50%;
    left: 50%;
    width: 50px;
    height: 50px;
    margin-top: -25px; /* half the height */
    margin-left: -25px; /* half the width */
    position: absolute;
  }
`;

export default Spinner;
