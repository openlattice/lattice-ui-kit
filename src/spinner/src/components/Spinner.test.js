import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Spinner from './Spinner';

describe('Spinner', () => {
  test('snapshot', () => {
    const wrapper = shallow(<Spinner />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
