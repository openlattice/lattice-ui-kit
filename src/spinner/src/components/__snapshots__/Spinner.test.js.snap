// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner snapshot 1`] = `
<Rotate
  className="fa-layers fa-fw fa-1x"
  duration="0.75s"
>
  <FontAwesomeIcon
    border={false}
    className=""
    color="#cdd1db"
    fixedWidth={false}
    flip={null}
    icon={
      Object {
        "icon": Array [
          512,
          512,
          Array [],
          "f111",
          "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm216 248c0 118.7-96.1 216-216 216-118.7 0-216-96.1-216-216 0-118.7 96.1-216 216-216 118.7 0 216 96.1 216 216z",
        ],
        "iconName": "circle",
        "prefix": "fal",
      }
    }
    id="spinner-circle"
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    symbol={false}
    title=""
    transform={null}
  />
  <FontAwesomeIcon
    border={false}
    className=""
    color="#555e6f"
    fixedWidth={false}
    flip={null}
    icon={
      Object {
        "icon": Array [
          512,
          512,
          Array [],
          "f3f4",
          "M460.115 373.846l-6.941-4.008c-5.546-3.202-7.564-10.177-4.661-15.886 32.971-64.838 31.167-142.731-5.415-205.954-36.504-63.356-103.118-103.876-175.8-107.701C260.952 39.963 256 34.676 256 28.321v-8.012c0-6.904 5.808-12.337 12.703-11.982 83.552 4.306 160.157 50.861 202.106 123.67 42.069 72.703 44.083 162.322 6.034 236.838-3.14 6.149-10.75 8.462-16.728 5.011z",
        ],
        "iconName": "spinner-third",
        "prefix": "fal",
      }
    }
    id="spinner-third"
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    symbol={false}
    title=""
    transform={null}
  />
</Rotate>
`;
